Ensure the current post does not appear in auto generated lists of backfilled posts

attackant opened the pull request:
Addresses #93 

Ensure the current post is not included in the any auto generated lists of backfilled posts (but the user can still choose it.) Also includes `npm audit fix` and a minor refactor for readability (hopefully)

attackant committed:
Exclude current post from backfillPosts response

This update changes the handling of the backfillPosts response in the wp-curate plugin. It introduces a check to filter out the currentPostId from the response array or return null if the response is an individual post matching the current post. This ensures that the current post is not included in the backfillPosts attribute.

attackant committed:
Update version and changelog

The version in wp-curate.php has been updated to 1.8.3. In this update, logic is also added to the backfillPosts query in the wp-curate plugin. This change is to ensure that the current post is not included in the array of backfilled posts, providing a more accurate and desired output.

mogmarsh commented:
I don't think this addresses the front end

mogmarsh commented:
I think to handle the front end, we should add a query class that will exclude the current post, then add that query class in https://github.com/alleyinteractive/wp-curate/blob/develop/src/features/class-query-block-context.php
I can try to help, or @dlh01 really knows this setup.

dlh01 commented:
The `Exclude_Queries` class in Type Extensions could be used, if you can supply the current post ID to it: https://github.com/alleyinteractive/wp-type-extensions/blob/a67070fc2d03422c5e24f83a211bffea7845ffa3/src/alley/wp/post-queries/class-exclude-queries.php

attackant commented:
> The `Exclude_Queries` class in Type Extensions could be used, if you can supply the current post ID to it: https://github.com/alleyinteractive/wp-type-extensions/blob/a67070fc2d03422c5e24f83a211bffea7845ffa3/src/alley/wp/post-queries/class-exclude-queries.php

Thanks! I will check it out, I was just trying to roll my own

attackant committed:
Update various npm package versions from npm audit fix

This commit includes an update to various npm package versions including 'body-parser', 'express', 'ip', 'follow-redirects', 'cookie', 'raw-body' and 'webpack-dev-middleware'. These updates are aimed at improving the overall functionality and security of the dependencies. Each library has been updated to a newer version, ensuring that we're using the latest available code.

attackant committed:
Refactor query block context in WP-Curate plugin

This commit overhauls the filter_query_context method in the WP-Curate WordPress plugin. It reorganizes code for improved readability and clarity

attackant committed:
Update comment

attackant committed:
Refactor Query_Block_Context to remove current post

mogmarsh commented on blocks/query/edit.tsx:
```diff
@@ -169,21 +170,31 @@ export default function Edit({
-        setAttributes({ backfillPosts: response as Array<number> });
+      apiFetch({ path }).then((response:any) => {
+        let revisedResponse;
+        if (Array.isArray(response)) {
+          revisedResponse = response.filter((item) => item !== currentPostId);
+        } else if (response.id === currentPostId) {
+          // If the current post is in the response, remove it.
+          revisedResponse = null;
+        } else {
+          revisedResponse = response;
+        }
```
the response should always be an array. I'm not sure how to convince typescript of that.

mogmarsh reviewed (approved):
I think this looks good. As far as I can tell the logic should be the same on the front end. Did you get the same results in the editor and the front end with this change? And did you get the same results on the front end with the current production and this PR (with the exception of removing the current post)?
I'm going to create an issue to add some tests so we can make changes like this with more confidence.

dlh01 commented on src/features/class-query-block-context.php:
```diff
@@ -20,99 +20,127 @@
  * Provides context to query blocks
  */
-final class Query_Block_Context implements Feature {
+final readonly class Query_Block_Context implements Feature {
```
As the CI results already noted, readonly classes aren't available while the plugin supports PHP 8.1.

dlh01 commented on CHANGELOG.md:
```diff
@@ -2,6 +2,10 @@
 
+## 1.8.3 - 2024-04-04
+
+- Enhancement: Exclude current post in backfilled posts query.
```
Since there are no bugfixes in the release, it seems more like a minor release than a patch release to me, i.e. `1.9.0`.

dlh01 reviewed (commented):
Updates on the queries side look good to me!

attackant committed:
Merge branch 'develop' into issue-93

attackant committed:
Merge branch 'develop' into issue-93

attackant committed:
Remove readonly and update properties

The class Query_Block_Context is no longer readonly to support PHP 8.1. Its properties have also been updated to readonly to prevent unintentional changes. An inline comment has been added for code clarification.

attackant committed:
Update WP Curate plugin version in changelog

The version of the WP Curate plugin has been updated in the changelog, from 1.8.3 to 1.9.0. The change reflects the latest enhancements including the exclusion of the current post in backfilled post queries.

attackant committed:
Update WP Curate plugin to version 1.9.0

This commit updates the WP Curate plugin to version 1.9.0 and makes minor changes to the code for better readability. It also shifts the add_action function and InstalledVersions class to the global namespace to eliminate the need for backslashes.

attackant committed:
Refactor query block edit.tsx for better error handling

This commit introduces a fallback case in the object destructuring assignment in query block edit.tsx to improve error handling. Additionally, it refines the parameter type in the useSelect hook for better type safety, making the logic clearer and securing the implementation.

attackant committed:
Add code comments

attackant committed:
Refine type assertion and function argument in edit.tsx

The commit includes a minor type assertion alteration and makes a slight adjustment in a function argument within 'edit.tsx'. The "(window as any as Window)" is modified to ensure that it correctly uses TypeScript type assertion. Also, "(select: any)" is updated in the 'useSelect' function, removing the unnecessary type info from the argument.

attackant committed:
Formatting only
